### ISSUE TITLE
'get_coro' doesn't apply to a 'Task' object

### ISSUE THREAD

**Issue by jacopo-exact** [agronholm/anyio#418]
Don't know if I should file it under `anyio` or `httpx`.

I have a FastAPI web app that makes some external calls with `httpx`. I sometimes (timeout involved? loop terminated elsewhere?) get the following error. I was unsuccessful at reproducing it in a minimal example, so please forgive me for just pasting the traceback.

As you can see in the first block, it is just a call with `httpx.AsycnClient` GET.

using 
python 3.9 (dockerized with python:3.9-alpine3.14)
anyio 3.5.0
httpx 0.19.0

EDIT: The problem could be related to the use of nest-asyncio elsewhere in the app.

```
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1740, in get
    return await self.request(
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1494, in request
    response = await self.send(
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1586, in send
    response = await self._send_handling_auth(
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1616, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1655, in _send_handling_redirects
    response = await self._send_single_request(request, timeout)
  File "/usr/local/lib/python3.9/site-packages/httpx/_client.py", line 1699, in _send_single_request
    ) = await transport.handle_async_request(
  File "/usr/local/lib/python3.9/site-packages/httpx/_transports/default.py", line 281, in handle_async_request
    ) = await self._pool.handle_async_request(
  File "/usr/local/lib/python3.9/site-packages/httpcore/_async/connection_pool.py", line 219, in handle_async_request
    async with self._connection_acquiry_lock:
  File "/usr/local/lib/python3.9/site-packages/httpcore/_backends/base.py", line 76, in __aenter__
    await self.acquire()
  File "/usr/local/lib/python3.9/site-packages/httpcore/_backends/anyio.py", line 104, in acquire
    await self._lock.acquire()
  File "/usr/local/lib/python3.9/site-packages/anyio/_core/_synchronization.py", line 119, in acquire
    self.acquire_nowait()
  File "/usr/local/lib/python3.9/site-packages/anyio/_core/_synchronization.py", line 150, in acquire_nowait
    task = get_current_task()
  File "/usr/local/lib/python3.9/site-packages/anyio/_core/_testing.py", line 59, in get_current_task
    return get_asynclib().get_current_task()
  File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 1850, in get_current_task
    return _create_task_info(current_task())  # type: ignore[arg-type]
  File "/usr/local/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 1846, in _create_task_info
    return TaskInfo(id(task), parent_id, name, get_coro(task))
TypeError: descriptor 'get_coro' for '_asyncio.Task' objects doesn't apply to a 'Task' object
```

**Comment by agronholm** [comment]
Are you using an alternative event loop? That error is nonsensical, saying Task's own `get_coro()` method does not apply to that class.

**Comment by jacopo-exact** [comment]
As I added in the edit, I was using `nest-asyncio` elsewhere in the code, and I have a feeling that is interfering. Other than that, I'm using vanilla asyncio.

To be more precise, I have a FastAPI backend with Motor for Mongodb support and httpx for http calls. `nest-asyncio` was used in a call to the Motor library, and caused no issue. The error above stems instead from a call to httpx, which always runs in an asyncio loop (i.e., without nest-asyncio nested loops).

I do agree with you that the error is nonsensical. That's precisely why I figured it was worth to report it.

**Comment by jacopo-exact** [comment]
Hi, here it is. 

Python 3.10.2

Deps:
```
anyio==3.5.0; python_full_version >= "3.6.2" and python_version >= "3.6"
certifi==2021.10.8; python_version >= "3.6"
charset-normalizer==2.0.11; python_full_version >= "3.5.0" and python_version >= "3.6"
h11==0.12.0; python_version >= "3.6"
htmlx==0.0.2; python_version >= "3.6"
httpcore==0.14.7; python_version >= "3.6"
httpx==0.22.0; python_version >= "3.6"
idna==3.3; python_version >= "3.6" or python_full_version >= "3.6.2" and python_version >= "3.6"
nest-asyncio==1.5.4; python_version >= "3.5"
rfc3986==1.5.0; python_version >= "3.6"
sniffio==1.2.0; python_full_version >= "3.6.2" and python_version >= "3.6" or python_version >= "3.6"
urllib3==1.26.8; python_version >= "3.6" and python_full_version < "3.0.0" or python_full_version >= "3.5.0" and python_version < "4" and python_version >= "3.6"
```


```
import asyncio
from httpx import AsyncClient
import nest_asyncio


async def http_call():
    async with AsyncClient() as client:
        res = await client.get("https://news.ycombinator.com")
    return res


def run_synchronously():
    nest_asyncio.apply()
    loop = asyncio.get_running_loop()
    res = loop.run_until_complete(http_call())
    return res


async def main():
    await http_call()
    run_synchronously()


if __name__ == "__main__":
    asyncio.run(main())
```

**Comment by jacopo-exact** [comment]
short answer is: I do not need it anymore. I removed every dependence on nest_asyncio from my code base.

As for why I needed it in the first place:
The specific case is that I need to give default values from a DB (managed fully async) to some pydandic models, using `Field(default_factory=<factory>)`.

The problem is that default_factory must be synchronous. To avoid having to maintain two implementations to access the DB, I was wrapping the async call to the DB within a synchronous function using nest-asyncio.

Lesson learnt. I took the time to reimplement the access to the db with native synchronous calls, and will be using the sync or async version where needed.

I still figured I would report it cause the error message makes no sense and it took me quite a while to figure out what was going on. This way the next me down the line that tries that abomination will find something that points to the way out.
